core/vm/gas_table.go

```diff
@@ -278,24 +278,59 @@ var (
 	gasMLoad   = pureMemoryGascost
 	gasMStore8 = pureMemoryGascost
 	gasMStore  = pureMemoryGascost
-	gasCreate  = pureMemoryGascost
 )
 
+func gasCreate(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
+	gas, err := memoryGasCost(mem, memorySize)
+	if err != nil {
+		return 0, err
+	}
+	len, overflow := stack.Back(2).Uint64WithOverflow()
+	if overflow {
+		return 0, ErrGasUintOverflow
+	}
+
+	if hasEIP3860(&evm.Config) {
+		lenWords := toWordSize(len)
+		initcodeWordGas, overflow := math.SafeMul(lenWords, params.InitCodeWordGas)
+		if overflow {
+			return 0, ErrGasUintOverflow
+		}
+		if gas, overflow = math.SafeAdd(gas, initcodeWordGas); overflow {
+			return 0, ErrGasUintOverflow
+		}
+	}
+	return gas, nil
+}
+
 func gasCreate2(evm *EVM, contract *Contract, stack *Stack, mem *Memory, memorySize uint64) (uint64, error) {
 	gas, err := memoryGasCost(mem, memorySize)
 	if err != nil {
 		return 0, err
 	}
-	wordGas, overflow := stack.Back(2).Uint64WithOverflow()
+	len, overflow := stack.Back(2).Uint64WithOverflow()
 	if overflow {
 		return 0, ErrGasUintOverflow
 	}
-	if wordGas, overflow = math.SafeMul(toWordSize(wordGas), params.Sha3WordGas); overflow {
+	lenWords := toWordSize(len)
+
+	hashingWordGas, overflow := math.SafeMul(lenWords, params.Sha3WordGas)
+	if overflow {
 		return 0, ErrGasUintOverflow
 	}
-	if gas, overflow = math.SafeAdd(gas, wordGas); overflow {
+	if gas, overflow = math.SafeAdd(gas, hashingWordGas); overflow {
 		return 0, ErrGasUintOverflow
 	}
+
+	if hasEIP3860(&evm.Config) {
+		initcodeWordGas, overflow := math.SafeMul(lenWords, params.InitCodeWordGas)
```

Comment:
Since the constant uses this capitalisation, maybe this is the standard in go?
```suggestion
		initCodeWordGas, overflow := math.SafeMul(lenWords, params.InitCodeWordGas)
```

Reply:
The constant was added by me, too, but changed variable name now for consistency.